contexts/security-v3-unstable.jsonld

```diff
@@ -5,11 +5,15 @@
         },
         "https://w3id.org/security/v2",
         {
-            "ethereumAddress" {
+            "owl": "https://www.w3.org/TR/owl-ref/#",
+            "ethereumAddress": {
                 "@id": "sec:ethereumAddress"
             },
             "publicKeyHex": {
                 "@id": "sec:publicKeyHex"
+            },
+            "sameAs": {
```

Comment:
Note that @talltree is currently adding this same property to the DID Core specification. Having it in the security vocabulary is dangerous because some developers might think it could be used to assert that a publicKey is the sameAs another public key. sameAs is an incredibly dangerous property in OWL land... it means that two things are semantically equivalent. So, if I were to say #orie sameAs #manu -- I would say that Orie is *exactly the same thing* as Manu, which is problematic, from a DID perspective. We are probably going to want to dial back the OWL semantics, possibly using something else to avoid the sledgehammer that is sameAs. We probably want something closer to "supersededBy" or "replacedBy" to match the use case that I think the community wants.

-0.8 to this PR until we can ferret out the use case.

/cc @iherman @dlongley

Reply:
well, its not assuming they are equivalent... the controller is saying the subjects are equivalent... if we don't trust the controller to make assertions about the subject... I fear we have made a mistake :)

Another thing to consider is that the strength of all RDF statements related to a subject is a function of the strength of the DID Method... so for example anyone with push access can add sameAs here: https://github.com/transmute-industries/did.actor/blob/master/alice/did.json#L3 ... whereas in did:v1, you would need private keys... however, assuming private key compromise is in some ways similar to assuming XSS... once its assumed, there are a whole bunch of other security issues... but in my opinion all of them flow from the XSS... similarly for private key compromise... the only reason you trust ANY assertion about the subject is that you believe that private key compromise HAS NOT occurred... once thats inverted, all the assertions (including sameAs) are worthless.

There is advantage, and danger is using OWL directly... but one thing for sure is that using OWL allows for OWL, whereas not using OWL is making something up that is new / not in use / not standardized...

Reply:
> Note that @talltree is currently adding this same property to the DID Core specification. Having it in the security vocabulary is dangerous because some developers might think it could be used to assert that a publicKey is the sameAs another public key (suggesting that all associated attributes can be merged together). [...]
> 
> To put it more concretely, one attack that sameAs enables is allowing the merging of compromised keys from an old DID to a new DID... because they're the same thing, right? (except they're not, and assuming they are opens us up to attack vectors).
> 
> -0.8 to this PR until we can ferret out the use case.
> 
> /cc @iherman @dlongley

I am not a security expert but I share the concern of @msporny. For example:

> well, its not assuming they are equivalent... the controller is saying the subjects are equivalent...  (@OR13)

Yes, we can/should trust the controller. The issue is that RDF/OWL is based on an open world assumption; this means that a `owl:sameAs` statement can be made by _anybody in the world_, it does not have to be the controller. It can be you, Manu, or me publishing this statement on the open Web. A bona fide OWL reasoning process that gets hold of this extra triple can make deductions that we may not want. 

As long as we remain within the DID world we are shielded from this because a DID processor is not expected to do any kind of OWL processing. I.e., this does not affect us. Introducing `owl:sameAs` _into_ the DID world gets us closer to OWL, even if minimally, and this may have unexpected consequences that we may not fully understand right now. I would just prefer to be cautious.

As far as I can see the goal is to use `owl:sameAs` in a very very constrained environment, and I personally would be o.k. to introduce our very own `identicalKey` term (or whatever purposes we want to use `sameAs`). Our vocabulary is small, and the places where an identification may be necessary is even smaller, I would not think this would be very onerous.